scripts/validate-front-matter/script.py

```diff
@@ -0,0 +1,180 @@
+        fm = {}
+        with open(fpath, 'r') as input:
+            content = input.read()
+
+            if not content.endswith('\n'):
+                result[NO_TRAILING_NEWLINE].add(fpath)
+                # Can't parse reliably, so skipping further checks
+                continue
+
+            try:
+                fm = get_front_matter(content)
+            except Exception as e:
+                print(f'ERROR: page {fpath} has no front matter')
+        
+        if fm is None:
+            result[NO_FRONT_MATTER_FOUND].add(fpath)
+            continue
+        
+        # Collect all front matter keys
+        for key in fm:
+            top_attributes.add(key)
+
+        # Evaluate linkTitle
+        if 'linkTitle' in fm:
+            if len(fm['linkTitle']) > 40:
+                result[LONG_LINK_TITLE].add(fpath)
+        
+        # Evaluate title
+        if 'title' in fm:
+            if len(fm['title']) > 40 and not 'linkTitle' in fm:
+                result[NO_LINK_TITLE].add(fpath)
+            if len(fm['title']) > 100:
+                result[LONG_TITLE].add(fpath)
+        else:
+            result[NO_TITLE].add(fpath)
+        
+        if 'description' in fm:
+            if fm['description'] is None:
+                result[NO_DESCRIPTION].add(fpath)
+            elif len(fm['description']) < 70:
+                result[SHORT_DESCRIPTION].add(fpath)
+            elif len(fm['description']) > 300:
+                result[LONG_DESCRIPTION].add(fpath)
+        else:
+            result[NO_DESCRIPTION].add(fpath)
+
+        # Evaluate menu
+        if 'menu' in fm:
+            if 'linkTitle' not in fm:
+                result[NO_LINK_TITLE].add(fpath)
+            if 'weight' not in fm:
+                result[NO_WEIGHT].add(fpath)
+        
+        # Evaluate owner
+        if 'owner' in fm:
+            # TODO: check owner validity
+            if type(fm['owner']) != list:
+                result[INVALID_OWNER].add(fpath)
+            elif len(fm['owner']) == 0:
+                result[NO_OWNER].add(fpath)
+            else:
+                for o in fm['owner']:
+                    if not o.startswith('https://github.com/orgs/giantswarm/teams/'):
+                        result[INVALID_OWNER].add(fpath)
+        else:
+            if not fpath.startswith(changes_path):
+                result[NO_OWNER].add(fpath)
+        
+        if 'user_questions' in fm:
+            for q in fm['user_questions']:
+                if len(q) > 80:
+                    result[LONG_USER_QUESTIONS].add(fpath)
+        else:
+            if not fpath.startswith(changes_path):
+                result[NO_USER_QUESTIONS].add(fpath)
+        
+        if 'last_review_date' in fm:
+            if type(fm['last_review_date']) is datetime.date:
+                diff = datetime.date.today() - fm['last_review_date']
+                if diff > datetime.timedelta(days=365):
+                    result[REVIEW_TOO_LONG_AGO].add(fpath)
```

Comment:
Maybe we could just bump the review date automatically

Reply:
I like the idea of bumping the review date automatically. But how can it be done? Would it get done every time the doc is opened? That wouldn't be accurate.

Reply:
I'd like to discuss this outside this PR.